src/DynamoApplications/PathManagerPreference.cs

```diff
@@ -0,0 +1,48 @@
+using Dynamo.Applications;
+using Dynamo.Configuration;
+using Dynamo.Core;
+using Dynamo.Interfaces;
+using Dynamo.Models;
+using Dynamo.PythonServices;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DynamoApplications
+{
+    public sealed class PathManagerPreference
+    {
+        internal static readonly Lazy<PathManagerPreference>
+            lazy =
+            new Lazy<PathManagerPreference>
+            (() => new PathManagerPreference());
+
+        public static PathManagerPreference Instance { get { return lazy.Value; } }
+
+        private PathManagerPreference()
+        {
+            IPathResolver pathResolver = CreateIPathResolver(false, "", "", "");
+            PathManager = new PathManager(new PathManagerParams
+            {
+                CorePath = string.Empty,
+                HostPath = string.Empty,
+                PathResolver = pathResolver
+            });
```

Comment:
It seems this singleton does nothing but initialize path manager? Can we make path manager itself Singleton class?

Reply:
> It seems this singleton does nothing but initialize path manager? Can we make path manager and PreferenceSetting itself Singleton class?

Done.